### ISSUE TITLE
Endless loop when using ReadAsync and skipping empty records

### ISSUE THREAD

**Issue by andyfurniss4** [JoshClose/CsvHelper#1208]
I've discovered what may be a small but with `CsvReader.ReadAsync`.

I wanted to ignore empty records, when I say empty I mean there's just delimiters and that's all on a record like so:

> A,B,C
> ,,
> 

There's also a completely empty row at the bottom of the CSV as shown above.

To accomplish this I used `csvReader.Configuration.ShouldSkipRecord = row => row.All(x => string.IsNullOrWhiteSpace(x))` which I saw was suggested on an old ticket here. Firstly, I had to add a second condition to the `ShouldSkipRecord` like so:  `csvReader.Configuration.ShouldSkipRecord = row => row == null || row.All(x => string.IsNullOrWhiteSpace(x))` as I was getting null at the end of the read (possibly that empty row at the end of my CSV?) which threw an exception.

When running my code, it seemed to just hang and after putting some breakpoints in I realised it was just hitting the `ShouldSkipRecord` callback over and over again. I did some digging around in the code for CsvReader and discovered that the first few lines of `Read` and `ReadAsync` differ slightly.

**ReadAsync:**

```
do
{
	context.Record = parser.Read();
}
while (context.Record != null && Configuration.ShouldSkipRecord(context.Record));
```

**Read:**

```
do
{
	context.Record = await parser.ReadAsync();
}
while (Configuration.ShouldSkipRecord(context.Record));
```

Wondering if it's because `ReadAsync` is missing the `context.Record != null` condition which `Read has?

**Comment by JoshClose** [comment]
Those two methods should be identical accept for async/await added to the async method. I'll fix that.

**Comment by JoshClose** [comment]
This is fixed and will be in version 12.1.1 on nuget.
